packages/pwa-kit-runtime/src/ssr/server/build-remote-server.js

```diff
@@ -473,6 +480,113 @@ export const RemoteServerFactory = {
         })
     },
 
+    /**
+     * @private
+     */
+    _setupRemoveBasePathFromPathMiddleware(app) {
+        // Use envBasePath as the feature flag for this middleware
+        // If envBasePath is `/`, '', or undefined when the server starts, we don't need to
+        // initialize this middleware.
+        if (getEnvBasePath()) {
+            const removeBasePathFromPath = (path) => {
+                const regex = new RegExp(`^${getEnvBasePath()}(/|$)`)
```

Comment:
Nit: Regex is hard, let's put a comment to explain it here. E.g match any path starts with envBasePath